common/src/main/java/com/microsoft/hydralab/performance/inspectors/AndroidBatteryInfoInspector.java

```diff
@@ -2,15 +2,47 @@
 // Licensed under the MIT License.
 package com.microsoft.hydralab.performance.inspectors;
 
+import com.microsoft.hydralab.common.util.ShellUtils;
+import com.microsoft.hydralab.common.util.TimeUtils;
 import com.microsoft.hydralab.performance.PerformanceInspection;
 import com.microsoft.hydralab.performance.PerformanceInspectionResult;
 import com.microsoft.hydralab.performance.PerformanceInspector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.Assert;
+
+import java.io.File;
 
 public class AndroidBatteryInfoInspector implements PerformanceInspector {
+    private static final String RAW_RESULT_FILE_NAME_FORMAT = "%s_%s_%s.txt";
+    protected Logger classLogger = LoggerFactory.getLogger(getClass());
 
     @Override
     public PerformanceInspectionResult inspect(PerformanceInspection performanceInspection) {
-        return null;
+        if (performanceInspection.isReset) {
+            initialize(performanceInspection);
+        }
+
+        File rawResultFolder = new File(performanceInspection.resultFolder, performanceInspection.appId);
+        Assert.isTrue(rawResultFolder.exists() || rawResultFolder.mkdir(), "rawResultFolder.mkdirs() failed in" + rawResultFolder.getAbsolutePath());
+        File rawResultFile = new File(rawResultFolder,
+                String.format(RAW_RESULT_FILE_NAME_FORMAT, getClass().getSimpleName(), performanceInspection.appId, TimeUtils.getTimestampForFilename()));
+
```

Comment:
Nitpicky:
There are 2 blank lines here, 1 blank line is better I think. :)

Reply:
Updated thx